corehq/blobs/sql_templates/setup_blobmeta_view.sql

```diff
@@ -0,0 +1,137 @@
+ALTER TABLE blobs_blobmeta RENAME TO blobs_blobmeta_tbl;
+
+CREATE OR REPLACE VIEW blobs_blobmeta AS
+/*
+ * Values in the "id" column are partitioned into two ranges:
+ *
+ * positive values: blob metadata records
+ * negative values: legacy form attachment metadata
+ *
+ * Legacy metadata can be updated and deleted, but new records should
+ * not be inserted into this range. Some legacy metadata fields cannot
+ * be updated: domain, type_code, created_on, expires_on.
+ */
+
+SELECT
+    "id",
+    "domain",
+    "parent_id",
+    "name",
+    "key",
+    "type_code",
+    "content_type",
+    "properties",
+    "created_on",
+    "expires_on",
+    "content_length"
+FROM blobs_blobmeta_tbl
+
+UNION ALL
+
+SELECT
+    -att."id" AS "id",
+    xform."domain",
+    att.form_id AS parent_id,
+    att."name",
+    (CASE
+        WHEN att.blob_bucket = '' THEN '' -- empty bucket -> blob_id is the key
+        ELSE COALESCE(att.blob_bucket, 'form/' || att.attachment_id)
+    END || att.blob_id)::VARCHAR(255) AS "key",
```

Comment:
This case statement causes a seq-scan on lookup by key so this will most likely need to be changed.

Reply:
Resolved by https://github.com/dimagi/commcare-hq/pull/21631/commits/ed3849de9fe9bca34f097dc2524e41d3b329e4b4